stl/inc/utility

```diff
@@ -927,6 +927,41 @@ _EXPORT_STD [[noreturn]] __forceinline void unreachable() noexcept /* strengthen
 #endif // defined(_DEBUG)
 }
 
+#ifdef __clang__
+// Clang 17 implements forward_like directly but is confused by our deduced return type
+template <int>
+struct _Dispatch_forward_like;
+template <>
+struct _Dispatch_forward_like<0> {
+    template <class _Ty>
+    using _Apply = _Ty&&;
+};
+template <>
+struct _Dispatch_forward_like<1> {
+    template <class _Ty>
+    using _Apply = _Ty&;
+};
+template <>
+struct _Dispatch_forward_like<2> {
+    template <class _Ty>
+    using _Apply = const _Ty&&;
+};
+template <>
+struct _Dispatch_forward_like<3> {
+    template <class _Ty>
+    using _Apply = const _Ty&;
+};
+
+template <class _Ty, class _Uty>
+using _Forward_like_t =
+    typename _Dispatch_forward_like<2 * is_const_v<remove_reference_t<_Ty>>

```

Comment:
Multiplying `int * bool` is weird. Can we simply use `bool, bool` specializations?

Reply:
Were this not a workaround I'd argue about throughput micro-optimization, but for workaround purposes `bool` × `bool` is fine.